hudi-hive/src/main/java/org/apache/hudi/hive/SchemaDifference.java

```diff
@@ -74,6 +67,17 @@ public boolean isEmpty() {
     return deleteColumns.isEmpty() && updateColumnTypes.isEmpty() && addColumnTypes.isEmpty();
   }
 
+  @Override
+  public String toString() {
+    return new StringJoiner(", ", SchemaDifference.class.getSimpleName() + "[", "]")
```

Comment:
It seems that we have changed the generated string pattern? Did you check whether the `SchemaDifference#toString` method is used in the key judgement logic or not?